libbeat/outputs/elasticsearch/client.go

```diff
@@ -200,23 +203,29 @@ func (client *Client) publishEvents(
 	// encode events into bulk request buffer, dropping failed elements from
 	// events slice
 	origCount := len(data)
+	span.Context.SetLabel("events_original", origCount)
 	data, bulkItems := bulkEncodePublishRequest(client.log, client.conn.GetVersion(), client.index, client.pipeline, data)
 	newCount := len(data)
+	span.Context.SetLabel("events_encoded", newCount)
 	if st != nil && origCount > newCount {
 		st.Dropped(origCount - newCount)
 	}
 	if newCount == 0 {
 		return nil, nil
 	}
 
-	status, result, sendErr := client.conn.Bulk("", "", nil, bulkItems)
+	status, result, sendErr := client.conn.Bulk(ctx, "", "", nil, bulkItems)
 	if sendErr != nil {
-		client.log.Errorf("Failed to perform any bulk index operations: %s", sendErr)
+		err := apm.CaptureError(ctx, fmt.Errorf("failed to perform any bulk index operations: %w", sendErr))
+		err.Send()
+		client.log.Error(err)
 		return data, sendErr
 	}
+	pubCount := len(data)
+	span.Context.SetLabel("events_published", pubCount)
 
 	client.log.Debugf("PublishEvents: %d events have been published to elasticsearch in %v.",
-		len(data),
+		pubCount,
 		time.Now().Sub(begin))
 
 	// check response for transient errors
```

Comment:
I see you are capturing `events_original`, `events_encoded`, and `events_published`. A few lines below we also calculate the number of failed events. Do you think it's worth capturing that number as well?

Reply:
great idea!